powerflex/helper/service_helper.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright (c) 2024 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helper
+
+import (
+	"context"
+	"strconv"
+	"terraform-provider-powerflex/powerflex/models"
+	"time"
+
+	"github.com/dell/goscaleio"
+	scaleiotypes "github.com/dell/goscaleio/types/v1"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// GetServiceState converts scaleiotypes.Template to models.Template
+func GetServiceState(input *scaleiotypes.ServiceResponse) models.ServiceResourceModel {
+	return models.ServiceResourceModel{
+		ID:                    types.StringValue(input.ID),
+		DeploymentName:        types.StringValue(input.DeploymentName),
+		DeploymentDescription: types.StringValue(input.DeploymentDescription),
+		Status:                types.StringValue(input.Status),
+		Nodes:                 types.Int64Value(int64(input.ServiceTemplate.ServerCount)),
+		Compliant:             types.BoolValue(input.Compliant),
+		TemplateID:            types.StringValue(input.ServiceTemplate.OriginalTemplateID),
+		TemplateName:          types.StringValue(input.ServiceTemplate.TemplateName),
+		FirmwareID:            types.StringValue(input.FirmwareRepository.ID),
+	}
+}
+
+// UpdateServiceState - function to update state file for Service resource.
+func UpdateServiceState(deploymentResponse *scaleiotypes.ServiceResponse, plan models.ServiceResourceModel) (models.ServiceResourceModel, diag.Diagnostics) {
+	state := plan
+	var diags diag.Diagnostics
+
+	state = GetServiceState(deploymentResponse)
+
+	state.DeploymentTimeout = plan.DeploymentTimeout
+
+	return state, diags
+}
+
+// UpdateServiceState - function to update state file for Service resource.
+func HandleServiceDeployment(ctx context.Context, deploymentResponse *scaleiotypes.ServiceResponse, plan models.ServiceResourceModel, gatewayClient *goscaleio.GatewayClient) (*scaleiotypes.ServiceResponse, diag.Diagnostics) {
+
+	var diags diag.Diagnostics
+
+	deploymentID := deploymentResponse.ID
+
+	couterForStopExecution := 0
+
+	var deploymentTimeout int
+
+	if !plan.DeploymentTimeout.IsNull() && plan.DeploymentTimeout.ValueInt64() > 0 {
+		deploymentTimeout, _ = strconv.Atoi(plan.DeploymentTimeout.String())
+	} else {
+		deploymentTimeout = 60 //Default
+	}
```

Comment:
```suggestion
	deploymentTimeout := plan.DeploymentTimeout.ValueInt64()
```
This would be enough because of the addition of defaults in the schema.